### PR TITLE
Add port UFO 50

### PR DESCRIPTION
**Pull request by JeodC**
# New Port for UFO 50

### CFW Tests
Ensure your game has been tested on all major CFWs:
- [x] AmberELEC
- [x] ArkOS
- [x] ROCKNIX
- [x] muOS
- [x] RPi3

### Resolution Tests
Test all major resolutions:
- [x] 480x320
- [x] 640x480
- [x] 720x720 (RGB30)
- [x] Higher resolutions (e.g., 1280x720)

### CODE REVIEW

File: ports/ufo50/UFO 50.sh [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
```
Comment (kloptops) [inline ports/ufo50/UFO 50.sh:16]:
```suggestion
```

No longer needed.

File: ports/ufo50/UFO 50.sh [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+get_controls
+
+# Setup permissions
+$ESUDO chmod 666 /dev/tty1
```
Comment (kloptops) [inline ports/ufo50/UFO 50.sh:21]:
```suggestion
```

Superfluous

File: ports/ufo50/UFO 50.sh [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+$ESUDO chmod 666 /dev/uinput
```
Comment (kloptops) [inline ports/ufo50/UFO 50.sh:22]:
```suggestion
```

Not needed.

File: ports/ufo50/UFO 50.sh [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+# Display loading splash
+if [ -f "$GAMEDIR/patchlog.txt" ]; then
+    $ESUDO ./libs/splash "splash.png" 1 
+    $ESUDO ./libs/splash "splash.png" 2000
+fi
+
+# Run the game
+$GPTOKEYB "gmloadernext" -c "./ufo50.gptk" & 
+./gmloadernext game.apk
```
Comment (kloptops) [inline ports/ufo50/UFO 50.sh:64]:
```suggestion

pm_platform_helper "$GAMEDIR/game.apk"
./gmloadernext game.apk
```

Missing pm_platform_helper call.

File: ports/ufo50/UFO 50.sh [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+# Kill processes
+$ESUDO kill -9 $(pidof gptokeyb)
```
Comment (kloptops) [inline ports/ufo50/UFO 50.sh:67]:
```suggestion
pm_finish
```

No longer needed, use `pm_finish` instead, next two lines are superfluous too.

**Review by kloptops — CHANGES_REQUESTED** [review]

**Review by kloptops — APPROVED** [review]

File: ports/ufo50/UFO 50.sh [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+# Display loading splash
+if [ -f "$GAMEDIR/patchlog.txt" ]; then
+    $ESUDO ./libs/splash "splash.png" 1 
+    $ESUDO ./libs/splash "splash.png" 2000
+fi
+
+# Run the game
+$GPTOKEYB "gmloadernext" -c "./ufo50.gptk" & 
+
+pm_platform_helper "$GAMEDIR/game.apk"
```
Comment (cdeletre) [inline ports/ufo50/UFO 50.sh:62]: I think it should be
```
pm_platform_helper "$GAMEDIR/gmloadernext"
```

File: ports/ufo50/UFO 50.sh [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+./gmloadernext game.apk
+
+# Kill processes
+pm_finish
+$ESUDO systemctl restart oga_events &
```
Comment (cdeletre) [inline ports/ufo50/UFO 50.sh:67]: `$ESUDO systemctl restart oga_events &` should be removed (replaced by `pm_finish`)

**Review by cdeletre — CHANGES_REQUESTED** [review]

**Review by cdeletre — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**